### PR TITLE
docker-image.yml for publishing the docker image on docker hub using git hub actions

### PR DESCRIPTION
**Pull request by vdeshkar**
Its uses to publish a Docker image on public docker hub account

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - name: Build the Docker image
+      run: docker build . --tag a2d2-build:latest
+   
+    - name: Login to Docker Hub
+      with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Publish the Docker image
+      run: docker push . --tag a2d2-build:latest
```
Comment (ddjain) [inline .github/workflows/docker-image.yml:24]: name of docker image will be **a2d2**

**Review by ddjain — APPROVED** [review]

**Review by marianbuenosayres — APPROVED** [review]

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+jobs:
+         
+  build-login-publish:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build the Docker image
+      run: docker build . --tag elimuinformatics/a2d2:latest
```
Comment (hankwallace) [inline .github/workflows/docker-image.yml:16]: The tag and push should also have a version - not just `latest`.

aka
```
docker build -t elimuinformatics/a2d2:1.0 -t elimuinformatics/a2d2:latest .
```

**Review by hankwallace — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-image.yml (hunk 2 above)
Comment (hankwallace) [reply .github/workflows/docker-image.yml:16]: This version should be updated each time code in here changes.

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -22,8 +22,14 @@ jobs:
...
-      - name: building docker image
-        run: docker build -t elimuinformatics/a2d2:${GITHUB_REF##*/} -t elimuinformatics/a2d2:latest .
-
-      - name: Publish the Docker image
-        run: docker push . --tag elimuinformatics/a2d2:${GITHUB_REF##*/} --tag elimuinformatics/a2d2:latest
+      - name: build image
+        run: docker build -t elimuinformatics/a2d2:latest .
+      
+      - name: tag image with commit id
+        run: docker image tag elimuinformatics/a2d2:latest elimuinformatics/a2d2:${GITHUB_REF##*/}
```
Comment (hankwallace) [inline .github/workflows/docker-image.yml:29]: We do NOT want to tag it with the commit id. The tags should be `latest` and a version number.

**Review by hankwallace — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-image.yml (hunk 3 above)
Comment (vdeshkar) [reply .github/workflows/docker-image.yml]: @hankwallace By mistake it was commit Id, It should be version id. So I have updated the PR

Comment (hankwallace) [reply .github/workflows/docker-image.yml]: What is "version id"?

Comment (vdeshkar) [reply .github/workflows/docker-image.yml]: In order to trigger github action we need to create a tag ID, which will be used as docker image version id.

**Review by hankwallace — APPROVED** [review]

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          
+      - name: build image
+        run: docker build -t elimuinformatics/a2d2:latest .
+        
+      - name: tag image with version id
+        run: docker image tag elimuinformatics/a2d2:latest elimuinformatics/a2d2:1.0.0
```
Comment (hankwallace) [inline .github/workflows/docker-image.yml:26]: Don't we also need to change the version number(s) in the pom.xml file(s)?

**Pull request merged**